jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

```diff
@@ -298,16 +311,18 @@ public JibContainerBuilder createJibContainerBuilder(
   @Override
   public List<Path> getDependencies() {
     List<Path> dependencies = new ArrayList<>();
-    FileCollection runtimeClasspath = getMainSourceSet().getRuntimeClasspath();
+    FileCollection runtimeClasspath = project.getConfigurations().getByName(configurationName);
     // To be on the safe side with the order, calling "forEach" first (no filtering operations).
-    runtimeClasspath.forEach(
-        file -> {
-          if (file.exists()
-              && file.isFile()
-              && file.getName().toLowerCase(Locale.US).endsWith(".jar")) {
-            dependencies.add(file.toPath());
-          }
-        });
+    runtimeClasspath
+        .getAsFileTree()
```

Comment:
`getAsFileTree()`: what does this do and why do we want to add this? Can you guarantee this does not change the current behavior? The order of the elements in the return value is important (that's why we have the code comment right above), but the [Javadoc](https://docs.gradle.org/current/javadoc/org/gradle/api/file/FileTree.html) of `FileTree` says

> The order of the files in a `FileTree` is not stable, even on a single computer.

Reply:
Well then we have to find something better for mocking, because implementation of configuration ist changing quite often

Reply:
And it's huge, I'm not into mocking a lot so I don't know how this would be done.

Reply:
Hmm... thanks for trying. I'll see what I can do. This can take some time though.

Reply:
Almost there. Only some minor last things blocking me.

Reply:
Heads-up: #3091 I may need to add or remove one or two, but it's working.

Reply:
#3091 is merged, Sync with HEAD, and you should be able to fix the tests.

Reply:
Integrated